### ISSUE TITLE
Allow for calling generated stored procedures from procedural logic

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#9170]
While all stored function calls can be modeled as expressions using `AbstractRoutine.asField()`, which is called by generated convenience methods, producing a `Field<T>`, this currently cannot be done for stored procedures.

We need new generated code that produces convenience API in generated `Routines` classes and package classes for each stored procedure, taking `Field<T>` arguments for `IN` parameters, and `Variable<T>` arguments for `OUT` and `IN/OUT` parameters.

This might depend on #9169 (`DSL::call`)

**Comment by lukaseder** [comment]
Duplicate of https://github.com/jOOQ/jOOQ/issues/11363
